subsys/secure_storage/src/psa/protected_storage.c

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2024
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <psa/protected_storage.h>
+#ifdef CONFIG_SECURE_STORAGE_PS_IMPLEMENTATION_CUSTOM
+#include <zephyr/secure_storage/ps.h>
+#else
+#include <zephyr/secure_storage/its.h>
+#define ITS_UID (secure_storage_its_uid_t){.uid = uid, \
+					   .caller_id = SECURE_STORAGE_ITS_CALLER_PSA_PS}
+#endif
+
+psa_status_t psa_ps_set(psa_storage_uid_t uid, size_t data_length, const void *p_data,
+			psa_storage_create_flags_t create_flags)
+
+{
+#ifdef CONFIG_SECURE_STORAGE_PS_IMPLEMENTATION_CUSTOM
```

Comment:
Instead of these `#ifdef` in C source files, why don't we use the same approach as for `aead.c/aead_get.c`, i.e. we include `protected_storage.c` only if `CONFIG_SECURE_STORAGE_PS_IMPLEMENTATION_CUSTOM` is not defined? IMO it would be nice to keep the same approach over the subsys.